sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowingStrategies.java

```diff
@@ -122,8 +126,13 @@ public static ClosingBehavior fromProto(RunnerApi.ClosingBehavior proto) {
   }
 
   // This URN says that the coder is just a UDF blob the indicated SDK understands
+  // TODO: standardize such things
   private static final String CUSTOM_CODER_URN = "urn:beam:coders:custom:1.0";
 
+  // This URN says that the WindowFn is just a UDF blob the indicated SDK understands
+  // TODO: standardize such things
+  private static final String CUSTOM_WINDOWFN_URN = "urn:beam:windowfn:custom:1.0";
+
   private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
 
   public static RunnerApi.MessageWithComponents toProto(WindowFn<?, ?> windowFn)
```

Comment:
I think it is an implementation detail. Later, coders will refer to other coders by reference and who knows what else will occur. Either way, it is just required to be a valid message + components.

But, indeed, there should be javadoc and I have added it. I actually thought that sort of thing failed precommit...